pyvo/dal/tap.py

```diff
@@ -306,6 +309,84 @@ def create_query(
         return TAPQuery(
             self.baseurl, query, mode, language, maxrec, uploads, self._session, **keywords)
 
+    def get_job(self, job_id):
+        """
+        Returns the job corresponding to an ID. Note that the caller must be
+        able to see the job in the current security context.
+
+        Parameters
+        ----------
+        job_id : str
+            ID of the job to view
+
+        Returns
+        -------
+        `~pyvo.io.vosi.endpoint.JobSummary` corresponding to the job ID
+        """
+        response = self._session.get(
+            self.baseurl + '/async/' + job_id,
+            stream=True)
+        response.raw.read = partial(response.raw.read,
+                                    decode_content=True)
+        return uws.parse_for_object(response.raw.read, uws.JobSummary)
+
+    def get_job_list(self, phases=None, after=None, last=None,
+                     short_description=True):
+        """
+        lists jobs that the caller can see in the current security context.
+        The list can be filtered on the server side by the phases of the jobs,
+        creation date time or
+        Note that by default jobs in 'ARCHIVED` phase are not returned.
+
+        Parameters
+        ----------
+        phases: list of str
+            Union of job phases to filter the results by.
+        after: datetime
+            Return only jobs created after this datetime
+        last: int
+            Return only the most recent number of jobs
+        short_description: flag - True or False
+            If True, the jobs in the list will contain only the information
+            corresponding to the TAP ShortJobDescription object (job ID, phase,
+            run ID, owner ID and creation ID) whereas if False, a separate GET
+            call to each job is performed for the complete job description.
+
+        Returns
+        -------
+        list of `~pyvo.io.vosi.endpoint.JobSummary`
+        """
+
+        query = ''
+        if phases:
+            for p in phases:
+                query += 'PHASE={}&'.format(p)
+
+        if after:
+            if isinstance(after, str):
+                after = datetime.strptime(after, IVOA_TIME_FORMAT)
+            query += 'AFTER={}&'.format(after.strftime(IVOA_TIME_FORMAT))
+
+        if last:
+            query += 'LAST={}'.format(last)
+
+        if query:
+            query = '?{}'.format(query.strip('&'))
+
+        response = self._session.get('{}/async{}'.format(self.baseurl, query),
```

Comment:
Requests will let you pass query params like this as a dict, which means you hopefully won't have to worry about this kind of string parsing and query building:

https://2.python-requests.org//en/master/user/quickstart/#passing-parameters-in-urls

Reply:
Doh! What was I thinking...